lib/src/builder.dart

```diff
@@ -98,7 +99,11 @@ class GeneratorBuilder extends Builder {
 
 Stream<GeneratedOutput> _generate(LibraryElement unit,
     List<Generator> generators, BuildStep buildStep) async* {
-  for (var element in getElementsFromLibraryElement(unit)) {
+  var elements = safeIterate(getElementsFromLibraryElement(unit), (e, [_]) {
```

Comment:
It seems like in general we could just handle this inside of `getElementsFromLibraryElement` or `_getElements` (whichever is throwing).